aten/src/ATen/native/native_functions.yaml

```diff
@@ -1026,6 +1026,9 @@
 
 - func: matmul_out(Tensor result, Tensor self, Tensor other) -> Tensor
 
+- func: matrix_diag(Tensor self, int64_t offset=0, int64_t dim1=-2, int64_t dim2=-1) -> Tensor
+  variants: function, method
```

Comment:
Should it be a method as well? (I'm not sure)

Reply:
I think moving from diagonal and back is quite common, so some people will prefer it to have a method. But if you think it's safer to just add the function until someone complains, I'll go with that.

Reply:
Sure let's keep it as this.